src/ray/common/asio/instrumented_io_context.cc

```diff
@@ -66,10 +66,9 @@ void instrumented_io_context::post(std::function<void()> handler,
   // GuardedHandlerStats synchronizes internal access, we can concurrently write to the
   // handler stats it->second from multiple threads without acquiring a table-level
   // readers lock in the callback.
-  boost::asio::io_context::post(
-      [handler = std::move(handler), stats_handle = std::move(stats_handle)]() {
-        RecordExecution(handler, std::move(stats_handle));
-      });
+  boost::asio::io_context::post([handler = std::move(handler), stats_handle]() mutable {
```

Comment:
Why do we need this change?

Reply:
On a second look, we definitely shouldn't need this change, since the move into the closure should prevent a deallocation of `stats_handle` from happening in the outer scope... I'll revert that commit and see if it still passes locally and in the CI.